rdr_service/resource/generators/participant.py

```diff
@@ -389,6 +389,30 @@ def _prep_participant(self, p_id, ro_session):
             'date_of_death': deceased_date_of_death
         }
 
+        # Collect participant pairing history
+        pairing = ro_session.query(ParticipantHistory.lastModified, ParticipantHistory.hpoId,
+                                   ParticipantHistory.organizationId, ParticipantHistory.siteId). \
+            filter(ParticipantHistory.participantId == p_id).order_by(ParticipantHistory.lastModified).all()
+
+        pairing_history = None
+        if pairing:
+            pairing_history = list()
+            for item in pairing:
+                hpo = ro_session.query(HPO.name).filter(HPO.hpoId == item.hpoId).first()
+                organization = ro_session.query(Organization.externalId). \
+                    filter(Organization.organizationId == item.organizationId).first()
+                pairing_history.append({
+                    'last_modified': item.lastModified,
+                    'hpo': hpo.name if hpo else None,
+                    'hpo_id': item.hpoId,
+                    'organization': organization.externalId if organization else None,
+                    'organization_id': item.organizationId,
+                    'site': self._lookup_site_name(p.siteId, ro_session),
```

Comment:
If there's a reason to get site values per history record instead the current `participant `table , we might be able to eliminate this if we join the info on the `pairing` query too?

Reply:
Per above, shouldn' t this be looking up the `item.siteId` instead pf `p.siteId`?    That's why I thought instead of calling the lookup method every time,  you could just get the site name from a join in the pairing query.

Reply:
Sorry, I misconstrued your comments.  This is fixed in the latest commits.